engine/table/src/test/java/io/deephaven/engine/table/impl/TestAggBy.java

```diff
@@ -818,7 +818,7 @@ public void testAggAllByWithFormatColumn() {
         assertEquals(2.0, cs.get(0));
 
         result = dataTable.formatColumns("Doubles=Decimal(`##0.00%`)").headBy(1);
-        Set<String> columnNames = result.getColumnSourceMap().keySet();
+        List<String> columnNames = result.getDefinition().getColumnNames();
```

Comment:
Why not use your new set method?

Reply:
test impl only checks size and iterates through it - no need for Set afaict.

Reply:
Sure. Just a question of an on-demand allocation vs. a cached one.

Reply:
Ah, I see the "concern"; this brings up the larger question of do we want / need to document the APIs in terms of whether they are cached or not / implementation details? I don't necessarily want to box us into defining `TableDefinition#getColumnNames()` as on-demand, `TableDefinition#getColumnNamesSet` as cached.

Reply:
It's not unreasonable IMO that we could make the case that TableDefinition impl should go from

```java
    private final List<ColumnDefinition<?>> columns;
    private Map<String, ColumnDefinition<?>> columnNameMap;
```

to

```java
    private final Map<String, ColumnDefinition<?>> columnNameMap;
```

. Majority of work is probably auditing usage of `io.deephaven.engine.table.TableDefinition#getColumns()` to see who actually _needs_ a List vs a Collection (which could be achieved efficiently by `columnNameMap.values()`).

Reply:
I suspect you're right that the vast majority of the time an ordered collection is sufficient. No need to scope creep this PR for that.